django/contrib/postgres/aggregates/general.py

```diff
@@ -49,17 +51,15 @@ class BoolOr(Aggregate):
 
 class JSONBAgg(OrderableAggMixin, Aggregate):
     function = "JSONB_AGG"
-    template = "%(function)s(%(distinct)s%(expressions)s %(ordering)s)"
     allow_distinct = True
     output_field = JSONField()
 
 
-class StringAgg(OrderableAggMixin, Aggregate):
-    function = "STRING_AGG"
-    template = "%(function)s(%(distinct)s%(expressions)s %(ordering)s)"
-    allow_distinct = True
-    output_field = TextField()
-
+class StringAgg(_StringAgg):
     def __init__(self, expression, delimiter, **extra):
-        delimiter_expr = Value(str(delimiter))
-        super().__init__(expression, delimiter_expr, **extra)
+        warnings.warn(
+            "The PostgreSQL specific StringAgg function is deprecated. Use "
+            "django.db.models.aggregate.StringAgg instead.",
+            category=RemovedInDjango60Warning,
+        )
+        super().__init__(expression, delimiter, **extra)
```

Comment:
@camuthig to make sure I understand, this is where you would add the logic to pass `Value(delimiter_expr)` to `super()` with a deprecation warning?

Reply:
@charettes I just pushed a change with this logic. Yes, this is where I was planning to do it.